packages/openapi-v3/src/__tests__/unit/decorators/request-body/request-body.spike.decorator.unit.ts

```diff
@@ -0,0 +1,223 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/openapi-v3
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  belongsTo,
+  Entity,
+  hasMany,
+  model,
+  property,
+} from '@loopback/repository';
+import {expect} from '@loopback/testlab';
+import {getControllerSpec, post, put} from '../../../..';
+import {requestBody2} from '../../../../decorators/request-body.spike.decorator';
+
+describe('spike - requestBody decorator', () => {
+  context('CRUD', () => {
+    @model()
+    class Product extends Entity {
+      @property({
+        type: 'string',
+      })
+      name: string;
+      @belongsTo(() => Category)
+      categoryId: number;
+
+      constructor(data?: Partial<Product>) {
+        super(data);
+      }
+    }
+
+    /**
+     * Navigation properties of the Product model.
+     */
+    interface ProductRelations {
+      category?: CategoryWithRelations;
+    }
+    /**
+     * Product's own properties and navigation properties.
+     */
+    type ProductWithRelations = Product & ProductRelations;
+
+    @model()
+    class Category extends Entity {
+      @hasMany(() => Product)
+      products?: Product[];
+    }
+    /**
+     * Navigation properties of the Category model.
+     */
+    interface CategoryRelations {
+      products?: ProductWithRelations[];
+    }
+    /**
+     * Category's own properties and navigation properties.
+     */
+    type CategoryWithRelations = Category & CategoryRelations;
+
+    it('create - generates schema with excluded properties', () => {
+      class MyController1 {
+        @post('/Product')
+        create(
+          @requestBody2(
+            {description: 'Create a product', required: true},
+            Product,
+            {exclude: ['id']},
+          )
+          product: Exclude<Product, ['id']>,
+        ) {}
+      }
+
+      const spec1 = getControllerSpec(MyController1);
+
+      const requestBodySpecForCreate =
+        spec1.paths['/Product']['post'].requestBody;
+
+      const referenceSchema = spec1.components!.schemas![
+        'ProductExcluding[id]'
+      ];
+
+      expect(requestBodySpecForCreate).to.have.properties({
+        description: 'Create a product',
+        required: true,
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ProductExcluding[id]',
+            },
+          },
+        },
+      });
+
+      // The feature that generates schemas according to
```

Comment:
Can we include this in the list of follow up stories in the markdown file you have (disregard if already included)?

Reply:
@b-admike Seems this feature is already supported.